lib/SILOptimizer/SILCombiner/SILCombinerBuiltinVisitors.cpp

```diff
@@ -89,6 +89,63 @@ SILInstruction *SILCombiner::optimizeBuiltinCanBeObjCClass(BuiltinInst *BI) {
   }
 }
 
+static unsigned getTypeWidth(SILType Ty) {
+  if (auto BuiltinIntTy =
+          dyn_cast<BuiltinIntegerType>(Ty.getSwiftRValueType())) {
+    if (BuiltinIntTy->isFixedWidth()) {
+      return BuiltinIntTy->getFixedWidth();
+    }
+  }
+  return 0;
+}
+
+SILInstruction *SILCombiner::optimizeBuiltinTruncOrBitCast(BuiltinInst *I) {
+  SILValue Op = I->getArguments()[0];
```

Comment:
Please assert that it is really a TruncOrBitCast builtin. Or may be extend the pattern to check it.